### PR TITLE
AndroidX

### PR DESCRIPTION
**Pull request by forkch**
migration to AndroidX

right now i cannot build due to some maven publishing taks which are not working on my machine (or due to the new gradle plugin..)

### CODE REVIEW

File: processor/build.gradle [hunk 1]
```diff
@@ -36,8 +36,8 @@ dependencies {
     testCompile "commons-io:commons-io:$COMMONS_IO_VERSION"
     testCompile androidJar()
     testCompile files(Jvm.current().getToolsJar())
-    testCompileAar "com.android.support:support-v4:$SUPPORT_LIBRARY_VERSION"
-    testCompileAar "com.android.support:support-v13:$SUPPORT_LIBRARY_VERSION"
+    testCompileAar 'androidx.legacy:legacy-support-v4:1.0.0-alpha3'
+    testCompileAar 'androidx.legacy:legacy-support-v13:1.0.0-alpha3'
```
Comment (shiraji) [inline processor/build.gradle]: Create a version code at `gradle.properties`

Comment (shiraji) [reply processor/build.gradle]: TO: @hotchemi @mannodermaus 

I was thinking we are still keep android support library's support. Do we want to drop the support of support library?

File: sample-kotlin/src/main/kotlin/permissions/dispatcher/samplekotlin/camera/CameraPreviewFragment.kt [hunk 2]
```diff
@@ -26,7 +24,7 @@ import permissions.dispatcher.samplekotlin.R
  * Implementation is based directly on the documentation at
  * http://developer.android.com/guide/topics/media/camera.html
  */
-class CameraPreviewFragment : Fragment() {
+class CameraPreviewFragment : androidx.fragment.app.Fragment() {
```
Comment (shiraji) [inline sample-kotlin/src/main/kotlin/permissions/dispatcher/samplekotlin/camera/CameraPreviewFragment.kt]: Why do not `import`?

**Comment by forkch** [comment]
I've fixed the mentioned issues (imports and versioning) and fixed the travis build by copying the licences to $ANDROID_HOME/licences

It is failing now due to the publishing plugin. I don't know how you usually deal with this.

**Comment by hotchemi** [comment]
Might be related to https://github.com/novoda/bintray-release could you try to update?

**Comment by forkch** [comment]
no luck...bintray-release does not seem to support gradle-4.6 but the android-gradle plugin for AndroidX should be the 3.2.0-alpha* 

https://github.com/novoda/bintray-release/issues/177

**Comment by hotchemi** [comment]
Umm.. 🤔 
I found kind of official plugin and seems there is a PR for that but haven't merged yet.
https://github.com/bintray/gradle-bintray-plugin/pull/230

**Comment by forkch** [comment]
Unfortunately I have no resources atm to look into bintray (have not yet deployed any artefacts to any kind of artifactory :-)).

**Comment by hotchemi** [comment]
Umm ok, it would be actually a huge change so it might be better to be addressed by @permissions-dispatcher/maintainers 🤔Really sorry to say that but this is actually a great work!

**Comment by shiraji** [comment]
Let me take over this work.

**Comment by shiraji** [comment]
@forkch if you don't mind, could I change your author mail address to your GitHub email address? This change lead to associate your commit with your GitHub account.

**Comment by forkch** [comment]
Sure no problem!

**Comment by shiraji** [comment]
Done. See #494

I will close this PR since all @forkch works are there. Thank you for your contribution!

**Comment by hotchemi** [comment]
Thank you @forkch !

**Pull request closed without merging**